### PR TITLE
Add INSTALL with build instructions

### PR DESCRIPTION
**Pull request by e271828-**

### CODE REVIEW

**Comment by luke-jr** [comment]
That looks generic enough to apply to any platform in one sense, and specific enough to be Homebrew only (and a particular configuration thereof) rather than general to OS X...

Also, it should probably create an `INSTALL` file rather than bundling in `README`.

**Comment by e271828-** [comment]
@luke-jr Fair enough. Updated PR as requested.

File: README [hunk 1]
```diff
@@ -54,3 +54,4 @@ Targetting base58check is done similarly to raw base58 encoding, but you must
 also provide a version byte:
 	bool b58check_enc(char *b58c, size_t *b58c_sz, uint8_t ver,
 	                  const void *data, size_t datasz)
+
```
Comment (luke-jr) [inline README]: Please remove :)

File: INSTALL [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Installation
+--------------------
+# Install libgcrypt first, e.g. via `brew install libgcrypt` on OS X or libgcrypt-dev on Linux
+# Set env var to libgcrypt path, for example on OS X with ver 1.7.6:
+export AM_PATH_LIBGCRYPT="/usr/local/Cellar/libgcrypt/1.7.6/lib"
+# Generate the final build scripts
+autoreconf --install
```
Comment (luke-jr) [inline INSTALL]: Should be `./autogen.sh`

File: INSTALL [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Installation
+--------------------
+# Install libgcrypt first, e.g. via `brew install libgcrypt` on OS X or libgcrypt-dev on Linux
+# Set env var to libgcrypt path, for example on OS X with ver 1.7.6:
+export AM_PATH_LIBGCRYPT="/usr/local/Cellar/libgcrypt/1.7.6/lib"
```
Comment (luke-jr) [inline INSTALL]: Normally this shouldn't be needed.

**Comment by e271828-** [comment]
@luke-jr I've made those changes, but added a troubleshooting section:
`./autogen.sh` didn't actually work for me on OS X, but `autoreconf --install` did. 
Similarly, libgcrypt wasn't picked up until I specified the env var.

**Comment by luke-jr** [comment]
What is the problem you had with `autogen.sh`?

`dist_doc_DATA` in `Makefile.am` needs updating.

**Comment by e271828-** [comment]
@luke-jr Wasn't producing a Makefile with buildable targets after ./configure. Since the obvious alternative worked I didn't investigate too closely. Have updated my build env since hitting that issue, and just tested that ./autogen.sh now works. Perhaps outdated autotools.

File: INSTALL [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+./configure && make
+
+Troubleshooting
+--------------------
+If libgcrypt isn't found after install, set AM_PATH_LIBGCRYPT env var to libgcrypt path 
+prior to running autogen.sh. For example, on OS X with ver 1.7.6:
+
+export AM_PATH_LIBGCRYPT="/usr/local/Cellar/libgcrypt/1.7.6/lib"
+
+If ./autogen.sh doesn't work, try upgrading autoconf and automake or running `autoreconf --install`
```
Comment (luke-jr) [inline INSTALL]: autogen runs this, so if there's no repeatable problem with autogen, let's leave this out.

File: Makefile.am [hunk 5]
```diff
@@ -14,7 +14,7 @@ pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libbase58.pc
 
 dist_noinst_SCRIPTS = autogen.sh
-dist_doc_DATA = AUTHORS COPYING README
+dist_doc_DATA = AUTHORS COPYING README INSTALL
```
Comment (luke-jr) [inline Makefile.am]: Try to keep this alphabetical order?

**Comment by luke-jr** [comment]
Looks good. Maybe squash?

**Comment by e271828-** [comment]
@luke-jr Squashed.

**Pull request merged**